repository.go

```diff
@@ -347,12 +348,64 @@ func PlainClone(path string, isBare bool, o *CloneOptions) (*Repository, error)
 // operation is complete, an error is returned. The context only affects to the
 // transport operations.
 func PlainCloneContext(ctx context.Context, path string, isBare bool, o *CloneOptions) (*Repository, error) {
+	dirEmpty := false
+	dirExist := false
+
+	file, err := os.Stat(path)
+	pathNotExist := os.IsNotExist(err)
```

Comment:
You can write here:

```
file, err := os.Stat(path)
if os.IsNotExist(err) {
  dirExist = true
} else if err != nil {
  return nil, err
}
```

If `os.Stat`returns an error and it is not a not exist error, something is wrong (e.g. permissions) and we should not continue.

Reply:
Sorry, my snippet was wrong.
Leave your logic as it is if behavior with permission errors is correct.
Do not use `pathNotExist` variable here only to use it for the if condition. Just write `if !os.IsNotExist(err)`.